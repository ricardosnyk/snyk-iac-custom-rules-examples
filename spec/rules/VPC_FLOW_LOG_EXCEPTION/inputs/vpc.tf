# Â© 2023 Snyk Limited All rights reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

provider "aws" {
  region = "us-east-1"
}

resource "aws_vpc" "valid" {
  cidr_block       = "10.0.0.0/16"
  instance_tenancy = "default"

  tags = {
    name = "cloudbank-fix"
  }
}

resource "aws_vpc" "valid2" {
  cidr_block       = "10.0.0.0/16"
  instance_tenancy = "default"

  tags = {
    name = "prod-vpc"
  }
}

resource "aws_flow_log" "valid2" {
  iam_role_arn    = aws_iam_role.example.arn
  log_destination = aws_cloudwatch_log_group.example.arn
  traffic_type    = "ALL"
  vpc_id          = aws_vpc.valid2.id
}



resource "aws_vpc" "invalid" {
  cidr_block       = "10.0.0.0/16"
  instance_tenancy = "default"

  tags = {
    name = "prod-vpc"
  }

}