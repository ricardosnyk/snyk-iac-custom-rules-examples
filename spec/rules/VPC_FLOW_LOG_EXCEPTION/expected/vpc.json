[
  {
    "passed": false,
    "ignored": false,
    "resource_id": "aws_vpc.invalid",
    "resource_namespace": "spec/rules/VPC_FLOW_LOG_EXCEPTION/inputs/vpc.tf",
    "resource_type": "aws_vpc",
    "severity": "high",
    "resources": [
      {
        "id": "aws_vpc.invalid",
        "type": "aws_vpc",
        "namespace": "spec/rules/VPC_FLOW_LOG_EXCEPTION/inputs/vpc.tf",
        "location": [
          {
            "filepath": "spec/rules/VPC_FLOW_LOG_EXCEPTION/inputs/vpc.tf",
            "line": 46,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "tags",
              "name"
            ],
            "location": {
              "filepath": "spec/rules/VPC_FLOW_LOG_EXCEPTION/inputs/vpc.tf",
              "line": 50,
              "column": 3
            }
          }
        ]
      }
    ]
  },
  {
    "passed": true,
    "ignored": false,
    "resource_id": "aws_vpc.valid",
    "resource_namespace": "spec/rules/VPC_FLOW_LOG_EXCEPTION/inputs/vpc.tf",
    "resource_type": "aws_vpc",
    "severity": "high",
    "resources": [
      {
        "id": "aws_vpc.valid",
        "type": "aws_vpc",
        "namespace": "spec/rules/VPC_FLOW_LOG_EXCEPTION/inputs/vpc.tf",
        "location": [
          {
            "filepath": "spec/rules/VPC_FLOW_LOG_EXCEPTION/inputs/vpc.tf",
            "line": 19,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "tags",
              "name"
            ],
            "location": {
              "filepath": "spec/rules/VPC_FLOW_LOG_EXCEPTION/inputs/vpc.tf",
              "line": 23,
              "column": 3
            }
          }
        ]
      }
    ]
  },
  {
    "passed": false,
    "ignored": false,
    "resource_id": "aws_vpc.valid2",
    "resource_namespace": "spec/rules/VPC_FLOW_LOG_EXCEPTION/inputs/vpc.tf",
    "resource_type": "aws_vpc",
    "severity": "high",
    "resources": [
      {
        "id": "aws_flow_log.valid2",
        "type": "aws_flow_log",
        "namespace": "spec/rules/VPC_FLOW_LOG_EXCEPTION/inputs/vpc.tf",
        "location": [
          {
            "filepath": "spec/rules/VPC_FLOW_LOG_EXCEPTION/inputs/vpc.tf",
            "line": 37,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "vpc_id"
            ],
            "location": {
              "filepath": "spec/rules/VPC_FLOW_LOG_EXCEPTION/inputs/vpc.tf",
              "line": 41,
              "column": 3
            }
          }
        ]
      },
      {
        "id": "aws_vpc.valid2",
        "type": "aws_vpc",
        "namespace": "spec/rules/VPC_FLOW_LOG_EXCEPTION/inputs/vpc.tf",
        "location": [
          {
            "filepath": "spec/rules/VPC_FLOW_LOG_EXCEPTION/inputs/vpc.tf",
            "line": 28,
            "column": 1
          }
        ],
        "attributes": [
          {
            "path": [
              "tags",
              "name"
            ],
            "location": {
              "filepath": "spec/rules/VPC_FLOW_LOG_EXCEPTION/inputs/vpc.tf",
              "line": 32,
              "column": 3
            }
          }
        ]
      }
    ]
  }
]